### PR TITLE
Remove catalog from web_backend/connections/list

### PR DESCRIPTION
**Pull request by pmossman**
## What
- Remove catalog from web_backend connection list response
- Add source/destination icons directly to connection response fields so that front-end doesn't need to call the source/destination-definitions endpoints

Will fill in detail later and add/fix tests, opening as a draft for now

### CODE REVIEW

File: airbyte-webapp/src/components/EntityTable/utils.tsx
Comment (timroes) [inline airbyte-webapp/src/components/EntityTable/utils.tsx:76]: :information_source: This no longer requires the source/destination definitions since we also made `icon` to be part of the response in the connections/list API in this PR.

**Review by krishnaglick — COMMENTED** [review]
I tested locally and can confirm icons load, as do connections. This is a moderate-sized change so I wouldn't mind more eyes. Or a walkthrough of the Java code perhaps, so I could feel more certain about everything.

File: airbyte-api/src/main/openapi/config.yaml
Comment (tealjulia) [inline airbyte-api/src/main/openapi/config.yaml:4602]: Is a status of inactive the same as what we call "disabled" in the frontend code?  I want to make sure we're still getting that with this call since the list view should disable the "launch" button (shows up if schedule is manual) if the connection is disabled.

Comment (pmossman) [reply airbyte-api/src/main/openapi/config.yaml:4602]: Yes, the `inactive` status is "disabled". All of these properties in the new `WebBackendConnectionListItem` are directly copied from the existing properties on `WebBackendConnectionRead`, so their usage should be exactly the same on the front-end

**Review by mfsiega-airbyte — APPROVED** [review]
LGTM!

**Comment by pmossman** [comment]
@timroes @krishnaglick @tealjulia this is ready to go from the back-end side of things. Before I merge I want at least one approval from the front-end team to cover the front-end changes here.

If anyone would like to walk through the back-end changes in here, I'm happy to hop on a zoom call tomorrow! 

Also, I deployed this to dev-3 and reserved it for 24 hours so if anyone would like to test these changes out in a cloud environment, you can use https://dev-3-cloud.airbyte.io/ to try things out

**Review by timroes — APPROVED** [review]
Tested once more and read once more over the code. LGTM

**Review by teallarson — APPROVED** [review]
tested connections list page on dev deploy with this.  lgtm.

**Pull request merged**
